b2sdk/bucket.py

```diff
@@ -354,6 +359,32 @@ def ls(
         if prefix != '' and not prefix.endswith('/'):
             prefix += '/'
 
+        # If we're running with wildcard-matching, we could get
+        # a different prefix from it.  We search for the first
+        # occurrence of any of the special characters and fetch
+        # parent path from that place.
+        # Examples:
+        #   'b/c/*.txt' –> 'b/c/'
+        #   '*.txt' –> ''
+        #   'a/*/result.[ct]sv' –> 'a/'
+        if with_wildcard:
+            for wildcard_character in ['*', '?', '[']:
+                if wildcard_character not in folder_to_list:
+                    continue
+
+                starter_index = folder_to_list.index(wildcard_character)
```

Comment:
here you are doing a O(N) lookup twice. I think it might be better to retrieve the index and catch ValueError if it's not present... but maybe I'm wrong and that's ugly?